app/code/Magento/Catalog/Block/Product/View/Attributes.php

```diff
@@ -82,9 +82,11 @@ public function getAdditionalData(array $excludeAttr = [])
         foreach ($attributes as $attribute) {
             if ($attribute->getIsVisibleOnFront() && !in_array($attribute->getAttributeCode(), $excludeAttr)) {
                 $value = $attribute->getFrontend()->getValue($product);
-
+                
                 if (!$product->hasData($attribute->getAttributeCode())) {
                     $value = __('N/A');
+                } elseif (is_array($value)) {
+                    continue;
```

Comment:
No-no, your initial idea with skipping everything in case of array was fine, `continue` here is a bit confusing.

Please move it upper:
```
if (is_array($value = $attribute->getFrontend()->getValue($product))) {
    continue;
}
```

and the rest will remain the same. Please squash changes into single commit when builds are green (you can ignore functional tests on Travis, they are failing sometimes due to timeouts).

Reply:
fix it